### ISSUE TITLE
Conflict with using "await" keyword in function argument position inside condition

### ISSUE THREAD

**Issue by Syynth** [formium/tsdx#190]
### Current Behavior

Certain usages of await keyword break babel plugin. Here is the minimum repro I could come up with:
```javascript
async function test(callback) {
  if (true) {
    return callback(await 0);
  }
  return callback();
}
```
which results in the following error:
```
✖ Failed to compile
(babel plugin) TypeError: Cannot read property 'await' of null
TypeError: Cannot read property 'await' of null
```
It happens in https://github.com/babel/babel/blob/master/packages/babel-traverse/src/path/lib/virtual-types.js#L153
after being called from https://github.com/rpetrich/babel-plugin-transform-async-to-promises/blob/master/async-to-promises.ts#L3217
This may be a bug in the babel plugin but it seems to be only caused when used in conjunction with some other plugin in the tsdx babel configuration.

### Expected behavior
It should compile the code correctly, as shown in [this](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=IYZwngdgxgBAZgV2gFwJYHsI2QUxMgCimABsSAjYKAawEoYBvAKBhlThgOQCcEd7mrVtxzIE3LMTKUaBYAHdgqZDAAMtANwsYAX20ixEmFIpVqBTUx1A&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.5.5&externalPlugins=babel-plugin-async-to-promises%401.0.5) Babel Repl

### Suggestions
I don't know if this is an issue with the plugin itself, in which case it would be ideal to get it fixed there, but in the mean time either the combinations of plugins which causes this would hopefully be identified or a different plugin for transforming async calls could be used.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| TSDX           | 0.9.0
| TypeScript       | 3.5.3
| Browser          | N/A
| npm/Yarn         | 6.9.0
| Operating System | Ubuntu 19.04

**Comment by sw-yx** [comment]
seems gnarly! i’m not even familiar with why we need babel-plugin-transform-async-to-promises, cant really help with this one, sorry.

**Comment by agilgur5** [comment]
Per https://github.com/formium/tsdx/pull/795#issuecomment-694738942, have added a regression test for this in #795 so can confirm that PR will fix this issue too.

> [This issue] was different from the other [`async-to-promises` issues](https://github.com/formium/tsdx/labels/topic%2Fasync-to-promises) in that it wasn't a correctness issue but an actual Babel plugin error, and per OP [h]ere, potentially due to interaction with other plugins in TSDX

**Comment by agilgur5** [comment]
So a similar "error, not correctness" bug was found in #869 just now, so I actually reproduced that and confirmed it as an upstream issue.

So I thought this is also an upstream issue, however, I was unable to reproduce this given the code from OP. So perhaps changes in other Babel plugins or the ordering or ones included have already resolved this. This report is from TSDX v0.9, so it's quite outdated now.

@jaredpalmer since you said you think you hit this, could you please provide a reproduction? Otherwise I think this is already resolved without #795

**Comment by agilgur5** [comment]
Just tried this on TSDX v0.9.0 and TS 3.5.3 as per OP's environment and similarly was not able to reproduce... So even on the same version this repro doesn't seem to work 😕
